app/src/main/java/com/conupods/OutdoorMaps/Services/OutdoorDirectionsService.java

```diff
@@ -0,0 +1,128 @@
+package com.conupods.OutdoorMaps.Services;
+
+import android.app.Activity;
+import android.os.Handler;
+import android.os.Looper;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.conupods.R;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Polyline;
+import com.google.android.gms.maps.model.PolylineOptions;
+import com.google.maps.DirectionsApiRequest;
+import com.google.maps.GeoApiContext;
+import com.google.maps.PendingResult;
+import com.google.maps.internal.PolylineEncoding;
+import com.google.maps.model.DirectionsResult;
+import com.google.maps.model.DirectionsRoute;
+import com.google.maps.model.TravelMode;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+
+public class OutdoorDirectionsService {
+
+    private GoogleMap mMap;
+    private GeoApiContext gac;
+
+    private Activity passedActivity;
+
+    private DirectionsResult directionsResult = new DirectionsResult();
+    private List<DirectionsRoute> mDirectionRoutes = new ArrayList<>();
+
+    private int timesEntered = 0;
+
+    public OutdoorDirectionsService(Activity activity, GoogleMap mapFragment) {
+        passedActivity = activity;
+
+        gac = new GeoApiContext.Builder()
+                .apiKey(activity.getString(R.string.Google_API_Key))
+                .build();
+
+        mMap = mapFragment;
+    }
+
+    public void computeDirections(LatLng origin, LatLng destination, TravelMode mode) {
+
+        // this sends the request to the api
+        DirectionsApiRequest directions = new DirectionsApiRequest(gac);
+
+        // to see other options
+        //directions.alternatives(true);
+
+        directions.origin(new com.google.maps.model.LatLng(origin.latitude, origin.longitude));
+        directions.destination(new com.google.maps.model.LatLng(destination.latitude, destination.longitude));
```

Comment:
dont you love how uniform google is

Reply:
its honestly super annoying that I couldnt remove them...